### PR TITLE
doc: mark CustomCodec as deprecated

### LINKED ISSUE #3694: Add grpc.ForceCodec-like ServerOption based on encoding.Codec
### What version of gRPC are you using?
google.golang.org/grpc v1.29.1

### What version of Go are you using (`go version`)?
go1.14.3 darwin/amd64

### What did you do?
Following up on https://github.com/grpc/grpc-go/issues/2189, the server side was missed, it is still using the deprecated `grpc.Codec` interface.

### What did you expect to see?

```go

// CustomCodec returns a ServerOption that sets a codec for message marshaling and unmarshaling.
//
// This will override any lookups by content-subtype for Codecs registered with RegisterCodec.
func CustomCodec(codec encoding.Codec) ServerOption {
	return newFuncServerOption(func(o *serverOptions) {
		o.codec = codec
	})
}
```

### What did you see instead?

```go

// CustomCodec returns a ServerOption that sets a codec for message marshaling and unmarshaling.
//
// This will override any lookups by content-subtype for Codecs registered with RegisterCodec.
func CustomCodec(codec Codec) ServerOption {
	return newFuncServerOption(func(o *serverOptions) {
		o.codec = codec
	})
}



// Codec defines the interface gRPC uses to encode and decode messages.
// Note that implementations of this interface must be thread safe;
// a Codec's methods can be called from concurrent goroutines.
//
// Deprecated: use encoding.Codec instead.
type Codec interface {
	// Marshal returns the wire format of v.
	Marshal(v interface{}) ([]byte, error)
	// Unmarshal parses the wire format into v.
	Unmarshal(data []byte, v interface{}) error
	// String returns the name of the Codec implementation.  This is unused by
	// gRPC.
	String() string
}
```

### PR DESCRIPTION
**Pull request by menghanl**
fixes #3694

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -242,6 +242,10 @@ func KeepaliveEnforcementPolicy(kep keepalive.EnforcementPolicy) ServerOption {
 // CustomCodec returns a ServerOption that sets a codec for message marshaling and unmarshaling.
 //
 // This will override any lookups by content-subtype for Codecs registered with RegisterCodec.
+//
+// Deprecated: server side codec can be registered, and it will be picked by the
```
Comment (dfawley) [inline server.go:246]: `Deprecated: register codecs using encoding.RegisterCodec.  The server will automatically use registered codecs based on the incoming requests' headers` ?

Comment (dfawley) [reply server.go:246]: ...`see also <link>`

Comment (menghanl) [reply server.go]: Done

**Comment by menghanl** [comment]
I updated vet to ignore this. It was already ignoring other codec related deprecated APIs.

File: server.go [hunk 2]
```diff
@@ -242,6 +242,11 @@ func KeepaliveEnforcementPolicy(kep keepalive.EnforcementPolicy) ServerOption {
 // CustomCodec returns a ServerOption that sets a codec for message marshaling and unmarshaling.
 //
 // This will override any lookups by content-subtype for Codecs registered with RegisterCodec.
+//
+// Deprecated: register codecs using encoding.RegisterCodec. The server will
+// automatically use registered codecs based on the incoming requests' headers.
+// See also
+// https://github.com/grpc/grpc-go/blob/master/Documentation/encoding.md#using-a-codec.
```
Comment (dfawley) [inline server.go:249]: Can you look up & see whether this is something we must keep for 1.x or if it's something we may delete in the future?  And add that to the comment here.  (See other "deprecated" comments for examples.)

Comment (menghanl) [reply server.go:249]: Done. All deprecated APIs in this file were all available at GA.

Comment (dfawley) [reply server.go:249]: Thanks.

**Pull request merged**